cloud/azure/database/mysql/mode/cpu.pm

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2021 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package cloud::azure::database::mysql::mode::cpu;
+
+use base qw(cloud::azure::custom::mode);
+
+use strict;
+use warnings;
+
+sub get_metrics_mapping {
+    my ($self, %options) = @_;
+
+    my $metrics_mapping = {
+        'cpu_percent' => {
+            'output' => 'CPU percent',
+            'label'  => 'cpu-usage',
+            'nlabel' => 'azmysql.cpu.utilization.percentage',
+            'unit'   => '%',
+            'min'    => '0',
+            'max'    => '100'
+        }
+    };
+
+    return $metrics_mapping;
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'filter-metric:s'  => { name => 'filter_metric' },
+        'resource:s'       => { name => 'resource' },
+        'resource-group:s' => { name => 'resource_group' },
+        'resource-type:s'  => { name => 'resource_type', default => 'servers' }
```

Comment:
I don't think it's a good thing to set a default value here.  Discovery will provide a solution to figure it out. 

Prefer checking that the value is defined and not null in the `check_options` sub. (You have a ready to copy/paste example line 64 :))